scripts/qemu.py

```diff
@@ -129,17 +130,30 @@ def launch(self):
         '''Launch the VM and establish a QMP connection'''
         devnull = open('/dev/null', 'rb')
         qemulog = open(self._qemu_log_path, 'wb')
+        args = self._wrapper + [self._binary] + self._base_args() + self.args
         try:
             self._pre_launch()
-            args = self._wrapper + [self._binary] + self._base_args() + self._args
             self._popen = subprocess.Popen(args, stdin=devnull, stdout=qemulog,
                                            stderr=subprocess.STDOUT, shell=False)
             self._post_launch()
         except:
+            self._load_io_log()
             if self.is_running():
                 self._popen.kill()
                 self._popen.wait()
-            self._load_io_log()
+            else:
+                exc_type, exc_value, exc_traceback = sys.exc_info()
+                msg = ("Error launching VM.\n"
+                       "Original Exception: \n%s\n"
+                       "Command:\n%s\n"
+                       "Output:\n%s\n" %
+                       (''.join(traceback.format_exception(exc_type,
+                                                           exc_value,
+                                                           exc_traceback)),
```

Comment:
Yes, but it starts with offset 0. Using your example from patch series:

```
    Traceback (most recent call last):
      File "<stdin>", line 1, in <module>
      File "qemu.py", line 156, in launch
        raise RuntimeError(msg)
    RuntimeError: Error launching VM.
    Original Exception:
    Traceback (most recent call last):
      File "qemu.py", line 138, in launch
        self._post_launch()
      File "qemu.py", line 122, in _post_launch
        self._qmp.accept()
      File "qmp/qmp.py", line 145, in accept
        self.__sock, _ = self.__sock.accept()
      File "/usr/lib64/python2.7/socket.py", line 206, in accept
        sock, addr = self._sock.accept()
    timeout: timed out

    Command:
    /usr/bin/qemu-system-aarch64 -chardev socket,id=mon,
    path=/var/tmp/qemu-23958-monitor.sock -mon chardev=mon,mode=control
    -display none -vga none
    Output:
    qemu-system-aarch64: No machine specified, and there is no default
    Use -machine help to list supported machines
```

vs:

```
    Traceback (most recent call last):
      File "<stdin>", line 1, in <module>
      File "qemu.py", line 156, in launch
        raise RuntimeError(msg)
    RuntimeError: Error launching VM.
    Original Exception:
        Traceback (most recent call last):
          File "qemu.py", line 138, in launch
            self._post_launch()
          File "qemu.py", line 122, in _post_launch
            self._qmp.accept()
          File "qmp/qmp.py", line 145, in accept
            self.__sock, _ = self.__sock.accept()
          File "/usr/lib64/python2.7/socket.py", line 206, in accept
            sock, addr = self._sock.accept()
    timeout: timed out
    Command:
        /usr/bin/qemu-system-aarch64 -chardev socket,id=mon,
        path=/var/tmp/qemu-23958-monitor.sock -mon chardev=mon,mode=control
        -display none -vga none
    Output:
        qemu-system-aarch64: No machine specified, and there is no default
        Use -machine help to list supported machines
```

(+- line)